app/controllers/news_controller.rb

```diff
@@ -6,6 +6,11 @@ class NewsController < ApplicationController
 
   def index
     @articles = NewsArticle.published.by_release_date.by_published_at
+
+    if params[:section].present?
+      set_section
+      @articles = @articles.by_section(@section)
```

Comment:
Does `:by_section` sort by date etc?
Should the other assign to @articles be in an else block so the news isn't loaded from the database twice?